cardano-sync/src/Cardano/Sync/LedgerState.hs

```diff
@@ -437,6 +446,16 @@ ledgerRewardUpdate env lsc =
       LedgerStateAllegra als -> Generic.allegraRewards (leNetwork env) als
       LedgerStateMary mls -> Generic.maryRewards (leNetwork env) mls
 
+-- We only compute 'AdaPots' for later eras. This is a time consuming
+-- function and we only want to run it on epoch boundaries, close to the chain tip.
```

Comment:
How is the "close to the chain tip" specified?

Reply:
`getAdaPots` returns `Just` only for Allegra and Mary.

Reply:
Can this information also be provided for Shelley (ie pre-Allegra) ? If so, we probably should.

Reply:
Yes it can, but it seems a somewhat expensive function, since it traverses the utxo in ledger state and epoch boundaries are already overloaded. It can be added easily though. For Byron, I haven't checked.

Reply:
We can drop it for Byron, ebcasue that will be mostly un-interesting. Worth doing for all Shelley related eras.